pippy/utils.py

```diff
@@ -224,3 +225,53 @@ def pp_group_barrier():
         run_func(my_pp_ranks, args, *extra_args)
 
     rpc.shutdown()
+
+
+def get_argparser(
+    default_schedule=None,
+    default_world_size=5,
+    default_rank=-1,
+    default_addr="localhost",
+    default_port="29500",
+    default_replicate="0",
+):
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--world_size",
+        type=int,
+        default=int(os.getenv("WORLD_SIZE", default_world_size)),
+    )
+    parser.add_argument(
+        "--rank", type=int, default=int(os.getenv("RANK", default_rank))
+    )
+    parser.add_argument(
+        "--master_addr",
+        type=str,
+        default=os.getenv("MASTER_ADDR", default_addr),
+    )
+    parser.add_argument(
+        "--master_port",
+        type=str,
+        default=os.getenv("MASTER_PORT", default_port),
+    )
```

Comment:
I am thinking if we should leave these four args in the user program, for two reasons:
1. These args may vary from program to program, so it may be hard to unify;
2. These args may come from environment variables, so it may be good for them to be in top level so that users know which env vars correspond to them.